lib/optimizely/decision_service.rb

```diff
@@ -417,6 +417,28 @@ def get_forced_variation(project_config, experiment_key, user_id)
       [variation, decide_reasons]
     end
 
+    def validated_forced_decision(context, user_context)
+      decision = user_context.get_forced_decision(context)
+      flag_key = context[:flag_key]
+      rule_key = context[:rule_key]
+      variation_key = decision ? decision[:variation_key] : decision
+      reasons = []
+      target = rule_key ? "flag (#{flag_key}), rule (#{rule_key})" : "flag (#{flag_key})"
+      if variation_key
+        variation = user_context.optimizely_client.get_flag_variation(flag_key, variation_key, 'key')
```

Comment:
I don't what happened to my previous comments on this line. github bug? :)
We need to change this line to same as other SDKs - pass 'config' as a parameter and move `get_flag_variation` to `config`, so we can remove dependency to `user_context.optimizely_client` here.